packages/@webex/plugin-cc/src/services/core/aqm-reqs.ts

```diff
@@ -1,22 +1,24 @@
 import {Msg} from './GlobalTypes';
 import * as Err from './Err';
 import {HTTP_METHODS, WebexRequestPayload} from '../../types';
-import {WebSocketManager} from './WebSocket/WebSocketManager';
 import HttpRequest from './HttpRequest';
 import LoggerProxy from '../../logger-proxy';
 import {CbRes, Conf, ConfEmpty, Pending, Req, Res, ResEmpty} from './types';
 import {TIMEOUT_REQ} from './constants';
+import {StateChange} from '../agent/types';
+import {AGENT_STATE_AVAILABLE} from '../../features/constants';
+import {EventBus} from './EventBus';
 
 export default class AqmReqs {
   private pendingRequests: Record<string, Pending> = {};
   private pendingNotifCancelrequest: Record<string, Pending> = {};
-  private webSocketManager: WebSocketManager;
   private httpRequest: HttpRequest;
+  private eventBus: EventBus;
 
-  constructor(webSocketManager: WebSocketManager) {
+  constructor() {
     this.httpRequest = HttpRequest.getInstance();
-    this.webSocketManager = webSocketManager;
-    this.webSocketManager.addEventListener('message', this.onMessage);
+    this.eventBus = EventBus.getInstance();
```

Comment:
Same logic here, no need to DI the web socket manager, it can just do the web socket related tasks. We can use the bus to manage and route the events.